### PR TITLE
Initial Classes

### PR DESCRIPTION
**Pull request by XedinUnknown**
Added `Exception` and `InvalidArgumentException` standards-compliant classes.

- [x] @XedinUnknown to add missing docs.

### CODE REVIEW

**Comment by XedinUnknown** [comment]
@mecha, could you review this PR please?

File: src/InvalidArgumentException.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    /**
+     * The argument associated with this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @var mixed
+     */
+    protected $argument;
+
+    public function __construct($message = '', $code = 0, RootException $previous = null, $argument = null)
```
Comment (mecha) [inline src/InvalidArgumentException.php]: Documentation required.

File: src/InvalidArgumentException.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    /**
+     * Assigns the argument.
+     *
+     * @since [*next-version*]
+     *
+     * @return $this
+     */
+    protected function _setArgument($argument)
```
Comment (mecha) [inline src/InvalidArgumentException.php]: Parameter doc missing.

**Review by mecha — CHANGES_REQUESTED** [review]

File: test/functional/InvalidArgumentExceptionTest.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        $argument = uniqid('argument-');
+        $subject = $this->createInstance($message, $code, $previous, $argument);
+
+        $this->assertInstanceOf('Exception', $subject, 'Subject is not a valid exception');
+        try {
+            throw $subject;
+        } catch (RootException $e) {
+            $this->assertInstanceOf(static::TEST_SUBJECT_CLASSNAME, $subject, 'A valid instance of the test subject could not be created');
+            $this->assertInstanceOf('Dhii\\Exception\\InvalidArgumentExceptionInterface', $subject, 'Subject is not a valid Dhii throwable');
+            $this->assertEquals($message, $e->getMessage(), 'Subject message is wrong');
```
Comment (mecha) [inline test/functional/InvalidArgumentExceptionTest.php]: I think an argument can be made that. given that this is a functional test, these assertions are not required as they have already been included in `ExceptionTest`.

Comment (XedinUnknown) [reply test/functional/InvalidArgumentExceptionTest.php]: I thought about that. But on the other hand, this is a different concrete class, with a different constructor, which sets the internal properties, and therefore needs to be tested separately.

**Comment by XedinUnknown** [comment]
- [ ] @XedinUnknown to add missing docs.

**Review by mecha — APPROVED** [review]
LGTM

**Pull request merged**